include/examples/simple_io_behavior.h

```diff
@@ -0,0 +1,146 @@
+#include "MotorLog.hpp"  // MotorLog
+#include "real_time_tools/timer.hpp"  // real_time_tools::Timer
+
+#include "kodlab_mjbots_sdk/io_behavior.h"  // kodlab::IOBehavior
+#include "kodlab_mjbots_sdk/log.h"  // logging macros
+#include "examples/simple_robot.h"  // SimpleRobot
+
+/**
+ * @brief Simple behavior that runs a positional PD control loop on the first
+ * joint in a SimpleRobot with gains adjustable via LCM input and motor state
+ * logged via LCM output
+ */
+class SimpleIOBehavior : public kodlab::IOBehavior<SimpleRobot,
+                                                   PDGains,
+                                                   MotorLog> {
+
+ public:
+
+  /**
+   * @brief Un-hides `robot_` from the `IOBehavior` namespace
+   * @note This is not necessary, but allows you to call `robot_` instead of
+   * `this->robot_`.
+   */
+  using kodlab::IOBehavior<SimpleRobot, PDGains, MotorLog>::robot_;
+
+  /**
+   * @brief Using the default `IOBehavior` constructor
+   */
+  using kodlab::IOBehavior<SimpleRobot, PDGains, MotorLog>::IOBehavior;
+
+  /**
+   * @brief Initialize class members and return true upon completion
+   * @return true if initialization successful, false otherwise
+   */
+  bool Init() override {
+    // Initializes desired torques to zero
+    desired_torques_ = std::vector<float>(robot_->joints.size(), 0);
+
+    // Initialize joint positions and velocities
+    joint_positions_ = robot_->GetJointPositions();
+    joint_velocities_ = robot_->GetJointVelocities();
+
+    // Return true to indicate successful initialization
+    return true;
+  }
+
+  /**
+   * @brief Begin the behavior
+   * @details Record the current position as a set point and start the internal
+   * behavior timer from zero.
+   * @param prev_behavior the previously active behavior
+   */
+  void Begin(const kodlab::Behavior<SimpleRobot> &prev_behavior) override {
+    // Begin realtime timer
+    time_now_ = 0.0;
+    timer_.tic();
+
+    // Set desired position as current position
+    desired_position_ = robot_->GetJointPositions()[0];
+  }
+
+  /**
+   * @brief Compute the PD-controller update and set robot torques
+   * @details This method updates the behavior timer, retrieves the latest joint
+   * state, and uses this information to compute desired control input via a
+   * proportional-derivative scheme using the member gains.
+   */
+  void Update() override {
+    // Update time
+    time_now_ = timer_.tac();
+
+    // Retrieve current joint positions
+    joint_positions_ = robot_->GetJointPositions();
+    joint_velocities_ = robot_->GetJointVelocities();
+
+    // Compute command torque for joint 0 via PD, set all others to zero
+    desired_torques_ = std::vector<float>(robot_->joints.size(), 0.0);
+    float pos_error = desired_position_ - joint_positions_[0];
+    float vel_error = DESIRED_VELOCITY - joint_velocities_[0];
+    desired_torques_[0] = gains.kp * pos_error + gains.kd * vel_error;
```

Comment:
At some point I should add my/a PID class.

Reply:
Yes please!